frontend/components/ClubEditPage/AddressField.tsx

```diff
@@ -0,0 +1,115 @@
+import { LoadScript } from '@react-google-maps/api'
+import { Libraries } from '@react-google-maps/api/dist/utils/make-load-script-url'
+import { ReactElement, useState } from 'react'
+import CreatableSelect from 'react-select/creatable'
+import usePlacesAutocomplete from 'use-places-autocomplete'
+
+import { BORDER, CLUBS_GREY, FOCUS_GRAY } from '../../constants/colors'
+
+const styles = {
+  control: ({ background, ...base }) => {
+    return {
+      ...base,
+      border: `1px solid ${BORDER}`,
+      boxShadow: 'none',
+    }
+  },
+  option: ({ background, ...base }, { isFocused, isSelected }) => {
+    const isEmphasized = isFocused || isSelected
+    return {
+      ...base,
+      background: isEmphasized ? FOCUS_GRAY : background,
+      color: CLUBS_GREY,
+    }
+  },
+}
+
+type AddressProps = {
+  addressValue: string
+  changeAddress: (state: string) => void
+}
+
+const AddressTypeaheadField = ({
+  changeAddress,
+  addressValue,
+}: AddressProps): ReactElement => {
+  const {
+    suggestions: { data },
+    setValue,
+  } = usePlacesAutocomplete({
+    requestOptions: {
+      location: { lat: () => 39.952104, lng: () => -75.193739 },
```

Comment:
I assume that this is the latitude and longitude for the University of Pennsylvania. Could be good to add a comment about the location so future people aren't confused!